edb_mustgather.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+# all logs collected are based off the EDB's troubleshooting guide
+# https://docs.enterprisedb.io/cloud-native-postgresql/1.15.1/troubleshooting/#troubleshooting
+
+EDB_CLUSTER_NAMESPACE=$1
+LOG_PATH=$2
+
+if [ -z "$1" ] || [ -z "$2" ]
+  then
+    echo "USAGE: edb_mustgather.sh <edb-cluster-namespace> <log path>"
+    echo "<edb-cluster-namespace>: Required - The namespace where the edb cluster is deployed in to"
+    echo "<log path>: Required - The path where the edb must gather will be stored"
+    exit 1
+fi
+
+EDB_OP_NAMESPACE='ibm-common-services'
+PG_OP=$(oc get po -n ${EDB_OP_NAMESPACE} -o=custom-columns=NAME:.metadata.name | grep postgresql-operator-controller-manager)
+EDB_CLUSTER_NAME=$(oc get cluster -n ${EDB_CLUSTER_NAMESPACE} -o=jsonpath='{.items[0].metadata.name}')
+EDB_POD_NAMES=$(kubectl get pod -l k8s.enterprisedb.io/cluster=${EDB_CLUSTER_NAME} -L role -n ${EDB_CLUSTER_NAMESPACE} -o=custom-columns=NAME:.metadata.name --no-headers)
+EDB_BACKUP_NAMES=$(kubectl get backups -l k8s.enterprisedb.io/cluster=${EDB_CLUSTER_NAME} -L role -n ${EDB_CLUSTER_NAMESPACE} -o=custom-columns=NAME:.metadata.name --no-headers)
+K8S_DATA="${TEMP_PATH}/kubernetes"
+K8S_NAMESPACES="${K8S_DATA}/namespaces"
+K8S_NAMESPACES_SPECIFIC="${K8S_NAMESPACES}/${EDB_OP_NAMESPACE}"
+K8S_NAMESPACES_EDB_DATA="${K8S_NAMESPACES_SPECIFIC}/edb"
+K8S_NAMESPACES_POD_DATA="${K8S_NAMESPACES_SPECIFIC}/pods"
+K8S_NAMESPACES_POD_DESCRIBE_DATA="${K8S_NAMESPACES_POD_DATA}/describe"
+K8S_NAMESPACES_POD_LOG_DATA="${K8S_NAMESPACES_POD_DATA}/logs"
+NS=${LOG_PATH}/namespaces
+SPECIFIC_NS_EDB_OP=${NS}/${EDB_OP_NAMESPACE}
+SPECIFIC_NS_CLUSTER=${NS}/${EDB_CLUSTER_NAMESPACE}
+PO=${SPECIFIC_NS_EDB_OP}/pods
+CLUSTER=${SPECIFIC_NS_CLUSTER}/cluster
+CLUSTER_PODS=${SPECIFIC_NS_CLUSTER}/pods
+CLUSTER_BACKUPS=${SPECIFIC_NS_CLUSTER}/backups
+
+
+mkdir ${NS}
+mkdir ${SPECIFIC_NS_EDB_OP}
+mkdir ${SPECIFIC_NS_CLUSTER}
+mkdir ${PO}
+mkdir ${PO}/${PG_OP}
+mkdir ${CLUSTER}
+mkdir ${CLUSTER}/${EDB_CLUSTER_NAME}
+mkdir ${CLUSTER_PODS}
+mkdir ${CLUSTER_BACKUPS}
+mkdir ${CLUSTER_BACKUPS}/${EDB_CLUSTER_NAME}
+
+function gatherEdbOperatorData() {
+    kubectl describe pod ${PG_OP} -n ${EDB_OP_NAMESPACE} > ${PO}/${PG_OP}/describe.txt
```

Comment:
@Conor-Quin would it be worth having all the EDB output under a single base folder instead of `${PO}` and `${CLUSTER}`, `${CLUSTER_PODS}` etc

Reply:
All logs for EDB are underneath a single base folder fed in to the script called `LOG_PATH`.

For example, if the script was called ` ./edb_mustgather.sh apic edb` the file paths will look like this
```
└── namespaces
    ├── apic
    │   ├── backups
    │   ├── cluster
    │   └── pods
    └── ibm-common-services
        └── pods
```

When called from within the generate_mustgather.sh the `LOG_PATH` will be `/kubernetes/namespaces/apic/edb` so everything would be underneath `/edb`